src/library/scala/concurrent/Future.scala

```diff
@@ -618,7 +664,7 @@ object Future {
    */
   def sequence[A, M[X] <: TraversableOnce[X]](in: M[Future[A]])(implicit cbf: CanBuildFrom[M[Future[A]], A, M[A]], executor: ExecutionContext): Future[M[A]] = {
     in.foldLeft(successful(cbf(in))) {
-      (fr, fa) => for (r <- fr; a <- fa) yield (r += a)
```

Comment:
Just trying to deconstruct this change, I'm always a bit wary of the "... and just changed a little detail in the implementation" parts of doc commits 😄 

If I read things correctly, this change does three things:

  - makes the code more concise
  - makes it more efficient for `failed` and Future.never`
  - uses `InternalCallbackExecutor`, for the `flatMap` (the outer generator in the for comprehension) while still using `executor` for the `map` (the inner generator).
  - uses `InternalCallbackExecutor` for the explicitly written `map` that remains in this method.

@viktorklang Is this list correct? What's the rationale? What changes for the user?